### PR TITLE
NFT-926: Ensure oracle is synced before interactions allowed

### PR DESCRIPTION
**Pull request by adamgobes**
https://user-images.githubusercontent.com/7279416/218160661-fdfb43b3-6e82-4f8e-a7b0-a07d31b9308d.mov

### CODE REVIEW

File: hooks/useVaultWrite/useVaultWrite.ts [hunk 1]
```diff
@@ -27,6 +29,10 @@ export function useVaultWrite(
 ) {
   const { address } = useAccount();
   const oracleInfo = useOracleInfo(OraclePriceType.lower);
+  const oracleInfoSynced = useOracleSync(
```
Comment (adamgobes) [inline hooks/useVaultWrite/useVaultWrite.ts:32]: should maybe rename the hook to useOracleSynced?

File: __tests__/hooks/useOracleSync.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+jest.mock('lib/chainHelpers', () => ({
+  getLatestBlockTimestamp: jest.fn(),
+}));
+
+const mockedGetLatestBlockTimestamp =
+  getLatestBlockTimestamp as jest.MockedFunction<
+    typeof getLatestBlockTimestamp
+  >;
+
+describe('useOracleSync', () => {
```
Comment (cnasc) [inline __tests__/hooks/useOracleSync.test.ts:32]: thank you for testing!

Comment (adamgobes) [reply __tests__/hooks/useOracleSync.test.ts:32]: client excellence 🖖

**Review by cnasc — APPROVED** [review]
LGTM

**Pull request merged**